### PR TITLE
Add "failed to refresh" toast notifications

### PR DESCRIPTION
**Pull request by hardingjam**
## Motivation

❗ First, merge #1738 

- Adds a Toast notification if Refreshing (invalidating) queries fails.

https://github.com/user-attachments/assets/1687d473-377f-4de7-9bd6-8dbf1bc54e92

## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs
- [x] included screenshots (if this involves a front-end change)

## Summary by CodeRabbit

- **New Features**
  - Added dedicated error toast notifications to Order Detail, Order Quote, Vault Detail, and Transactions Listener components for refresh failures and transaction errors.
  - Introduced a new error toast helper function for consistent error messaging across components.

- **Bug Fixes**
  - Improved error handling by displaying error toasts when manual refresh actions fail, providing clearer user feedback.

- **Tests**
  - Expanded test coverage for error toast notifications and refresh failure scenarios.
  - Refined test structure with consistent mocking of error toasts and query invalidation behaviors.
  - Added tests verifying error toast functionality and edge cases in toast management.

### CODE REVIEW

File: packages/ui-components/src/__tests__/VaultDetail.test.ts [hunk 1]
```diff
@@ -162,7 +180,26 @@ describe('VaultDetail', () => {
...
+
+		await waitFor(async () => {
+			const refreshButton = await screen.getByTestId('top-refresh');
+			await userEvent.click(refreshButton);
+			expect(invalidateTanstackQueries).toHaveBeenCalledWith(queryClient, [
+				'100',
+				QKEY_VAULT + '100'
+			]);
+		});
+	});
```
[the remark this answers is not in the document]
Comment (hardingjam) [reply packages/ui-components/src/__tests__/VaultDetail.test.ts:196]: I prefer string concatenation here as it's more readable. Also it mirrors the way in which these strings are constructed in usage.

File: packages/ui-components/src/__tests__/TanstackOrderQuote.test.ts [hunk 2]
```diff
@@ -1,185 +1,260 @@
...
-			orderbookAddress: '0x123',
-			handleQuoteDebugModal: vi.fn()
-		},
-		context: new Map([['$$_queryClient', queryClient]])
+describe('TanstackOrderQuote component', () => {
+	beforeEach(() => {
+		vi.clearAllMocks();
+		vi.resetAllMocks();
+		(useToasts as Mock).mockReturnValue({
+			toasts: writable([]),
```
[the remark this answers is not in the document]
Comment (hardingjam) [reply packages/ui-components/src/__tests__/TanstackOrderQuote.test.ts:33]: done

File: packages/ui-components/src/__tests__/TanstackOrderQuote.test.ts [hunk 3]
```diff
@@ -1,185 +1,260 @@
...
+		const refreshButton = screen.getByTestId('refresh-button');
+		fireEvent.click(refreshButton);
+
+		await waitFor(() => {
+			const orderQuoteRows = screen.getAllByTestId('bodyRow');
+
+			// Check ETH/USD row
+			expect(orderQuoteRows[0]).toHaveTextContent('ETH/USD');
+			expect(orderQuoteRows[0]).toHaveTextContent('5.945438972656012126');
+			expect(orderQuoteRows[0]).toHaveTextContent('6.305004957644166012');
```
[the remark this answers is not in the document]
Comment (hardingjam) [reply packages/ui-components/src/__tests__/TanstackOrderQuote.test.ts:151]: fixed

File: packages/ui-components/src/__tests__/TanstackOrderQuote.test.ts [hunk 4]
```diff
@@ -1,185 +1,260 @@
...
+vi.mock('$lib/providers/toasts/useToasts', () => ({
+	useToasts: vi.fn()
+}));
 
-	await waitFor(() => {
-		const orderQuoteComponent = screen.getByTestId('bodyRow');
+vi.mock('$lib/queries/queryClient', async (importOriginal) => ({
+	...((await importOriginal()) as object),
+	invalidateTanstackQueries: vi.fn()
+}));
```
[the remark this answers is not in the document]
Comment (hardingjam) [reply packages/ui-components/src/__tests__/TanstackOrderQuote.test.ts:24]: the pattern i've used is fine, it's working and is used in all our other mocks

File: packages/ui-components/src/lib/types/toast.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+// Define the Toast interface
+export interface ToastProps {
+	message: string;
+	type: 'success' | 'error' | 'warning' | 'info';
+	color: 'green' | 'red' | 'yellow' | 'blue';
```
[the remark this answers is not in the document]
Comment (hardingjam) [reply packages/ui-components/src/lib/types/toast.ts]: this offers more flexibility, so i will keep it for now

File: packages/ui-components/src/lib/components/detail/VaultDetail.svelte [hunk 6]
```diff
@@ -64,6 +65,18 @@
 	onDestroy(() => {
 		clearInterval(interval);
 	});
+
+	const handleRefresh = async () => {
+		try {
+			await invalidateTanstackQueries(queryClient, [id, QKEY_VAULT + id]);
+		} catch {
+			addToast({
```
Comment (findolor) [inline packages/ui-components/src/lib/components/detail/VaultDetail.svelte:73]: i think we should add a new toast function called `errToast` since the type and color for these toasts are the same. only the message changes. this goes for all the other addToast calls with error type

**Review by findolor — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/lib/components/detail/VaultDetail.svelte (hunk 6 above)
Comment (hardingjam) [reply packages/ui-components/src/lib/components/detail/VaultDetail.svelte]: Good idea, have updated

**Review by findolor — APPROVED** [review]

**Review by hardyjosh — APPROVED** [review]
